### ISSUE TITLE
EOReader 0.17.0 and SAR: BUG whith DSPK bands

### ISSUE THREAD

**Issue by TK12331** [sertit/eoreader#62]
When I followed the document of https://eoreader.readthedocs.io/en/latest/notebooks/SAR.html#load-bands, some problems occurred.

First, there are some bool values of  band_names returned from `s1_prod.has_band(band)` are True, while they don't exist in `get_existing_bands()`
![image](https://user-images.githubusercontent.com/76766406/204481125-1241ea66-0193-4aaa-850f-421f93262bdb.png)

Second, I check the output file and there are only two tif files
![image](https://user-images.githubusercontent.com/76766406/204482305-a9e67d6a-e472-4ac0-898b-3ccaac3ebe43.png)
which causes the code `s1_prod.load(ok_bands, resolution=20.)` to fail.

Are these normal phenomenon?

**Comment by TK12331** [comment]
For the question 2, is DSPK supposed to exist?

**Comment by remi-braun** [comment]
Yes, it creates a VV_DSPK file on disk.
This fails for you and I want to know why 😓 

I have corrected a bug in the latest version and you may have encountered it...

**Comment by TK12331** [comment]
I download this repository the day before yesterday, should I try to update the code?
I'm restarting running the code now, it may takes some time.

**Comment by remi-braun** [comment]
I would like you to give me the full bug stack trace if you can so I can analyze if 😄 

In order not to reprocess everything, you should set an output directory and leave `remove_tmp` in `Reader().open` to False (its default value).

```python
prod = Reader().open(path, output_path="your/path")
```

or 

```python
prod = Reader().open(path)
prod.output = "your/path"
```

**Comment by TK12331** [comment]
After `s1_prod = reader.open(s1_path, output_path=f"{s1_path[:-4]}/EO/")` I got VH and VV tif without throwing an error, can you tell me which code file should I set breakpoint at? (Or rather the code file that create the VH_DSPK tif)

**Comment by remi-braun** [comment]
Yes VV and VH will be OK.
I suspect the error to be thrown by SNAP during the despeckling conversion. (`sar_products._despeckle_sar`)
Can you try:

```python
s1_prod.load(VV_DSPK)
```
That should fail with something like https://forum.step.esa.int/t/exception-found-when-reading-compressed-tif/654/7

**Comment by TK12331** [comment]
I found that the self._despeckle_sar was skipped, thus didn't run into `_despeckle_sar()` .
![image](https://user-images.githubusercontent.com/76766406/204524827-bd7613d5-fd41-4155-bdc0-8e4e0354b669.png)

I set `dspk=True` in `_write_sar`, and DSPK can be wrote.
![image](https://user-images.githubusercontent.com/76766406/204525087-5fdb2ecb-4b4d-4f46-9276-b229a428ecd2.png)

Can I just set `dspk=True` in `_write_sar` ? Is it correct?

Besides, `s1_prod.load(VV_DSPK)` would thow 'file not found' error.

**Comment by remi-braun** [comment]
`_despeckle_sar` is skipped if you don't try load DSPK bands.

I don't understand what you are currently testing.
Can you be clearer ? For example show me your testing snippets ?

**Comment by TK12331** [comment]
Here is my testing code.
![image](https://user-images.githubusercontent.com/76766406/204690332-aaae4fd8-1a77-42ef-be44-64b97db8bf88.png)

in `get_band_paths`, the `sab.is_despeckle(band)` is False thus skip the `_despeckle_sar`
![image](https://user-images.githubusercontent.com/76766406/204690611-148111c3-c163-4413-9930-c2210b3ec20a.png)

**Comment by TK12331** [comment]
If you need more messages, plz make me know, I'm always online.

**Comment by remi-braun** [comment]
Hello,

The line `get_band_path` is useless for you, you can directly load the bands 😄 
But you will likely have the same issue.
Otherwise your code is OK.

Could you give me the SNAP error ? Or maybe it isn't displayed in notebooks...

What is your sertit version ? For that type:
```
import sertit
sertit.__version__
```

I feel all that revolves around an [already fixed bug in sar_products.py](https://github.com/sertit/eoreader/commit/083f12b9b3f1273bc7b6acbc06479823b91cefeb) (the fix will be in 0.18.0 version), that comes from the fact that VV and VH GeoTiff's are saved on disk with an optimized format (LZW + predictor = 3, added in `sertit==1.19.2`),  and sadly SNAP doesn't handle it...

If it's the case, you can fork EOReader and use the 0.18.0 branch until its final release or downgrade sertit to 1.19.1 😓

**Comment by TK12331** [comment]
The SNAP error isn't displayed...
The version of my sertit is '1.19.5'.
I'm trying the 0.18.0 branch, hope it will work.

By the way, here are some questions I want to ask:
1. Should operation of Multi_Look should be conducted when processing S1 product? I note that other programs seem to conduct Multi_Look before Terrain_Correction.

3. What are `footprint` and `extent` as follows mean? I have thought about it for a long time...
![image](https://user-images.githubusercontent.com/76766406/204748519-55324ce0-e73d-4737-9ec3-67983f8a4349.png)

4. Can I directly load tiff file by OpenCV in python for post SAR image processing? Because it seems totally black in vision now...
![image](https://user-images.githubusercontent.com/76766406/204750081-03e8b8a0-e75b-4d82-ab21-7882cc9fa7b8.png)
If not, how can I convert Tiff to PNG reasonably?

5. Why the value in the VV band I load (xarray type) is -60 to 25, the value seems strange due to the usual value is 0 to a positive number that depends on the precision ( etc, int16, int32)
![image](https://user-images.githubusercontent.com/76766406/204749529-a85409df-4e90-49c5-adeb-5ce41667e7d5.png)

Thank you in advance for your help !

**Comment by remi-braun** [comment]
1/ Multi-look is for SLC data, not GRDH. If you look at complex graphs in `eoreader.data` you will see MultiLook before Terrain-Correction 😄 
2/ The difference between extent and footprint is easier to understand [here](https://eoreader.readthedocs.io/en/v0.17.0/notebooks/optical.html). Basically, the footprint is the data polygon, when extent includes nodata.
3/ I never used OpenCV and I wouldn't recommend it as it isn't designed to deal with rasters. Moreover, jpg will compress your file and you will lose a lot of information.
4/ VV band is in float32 and in dB. SAR data can have negative values, especially when processed.

You may need to take some courses on SAR data if you want to truly understand what you are doing as this isn't really the place to ask about it, as I am no expert 😉

**Comment by remi-braun** [comment]
PS: if you don't manage to use 0.18.0 branch, downgrade sertit to 1.19.1 by doing `pip install sertit==1.19.1`

**Comment by TK12331** [comment]
Got it, I'm a new researcher in SAR image domain hhh. Thanks for your advice.

I just found that `pip install ` doesn't have 0.18.0 version
![image](https://user-images.githubusercontent.com/76766406/204753768-37791278-e974-4a2a-b68e-5f5eb86565ac.png)

And I meet some strange problem when running the code from 0.18.0 branch like 
![image](https://user-images.githubusercontent.com/76766406/204753913-078820cd-6d9c-4f63-8486-040dd13780ba.png)

I 'm using python==3.9 and eoreader==0.17.0 now, does the main branch is version 0.17.0? How can I update it to 0.18.0?

**Comment by remi-braun** [comment]
As I said, 0.18.0 version has not been release yet 😉 
Try the downgrade stated hereover 🙏

**Comment by TK12331** [comment]
I have downgraded the sertit, and a new problem is raised.
![1669800050221](https://user-images.githubusercontent.com/76766406/204757791-2c3d3ba8-927b-4419-84c6-5e5ddf2c263e.jpg)

**Comment by TK12331** [comment]
I have downgraded the sertit, and a new problem is raised.
![1669800050221](https://user-images.githubusercontent.com/76766406/204757791-2c3d3ba8-927b-4419-84c6-5e5ddf2c263e.jpg)

And error `something went wrong with snap ` appeared again...

**Comment by remi-braun** [comment]
Aaaah that is what I wanted to see in the first time ! SNAP traceback 🥂 

Did you recomputed VH and VV files ? Or used the computed with the other version ?
If not, you need to delete/recompute them as they are saved on disk with a compression SNAP is not able to read....

**Comment by TK12331** [comment]
I indeed not recomputed it and used the cache before, I try it right now (about 4mins).

**Comment by TK12331** [comment]
The DSPK file can be produced now, thank you!

**Comment by remi-braun** [comment]
Ouuuh yeah, so relieved! 🚀 
Sorry you arrived at a bad time 😅 

This will all be resolved in 0.18.0...

**Comment by remi-braun** [comment]
For other people that have the same issue.

**TLDR**:
Attempt to compute DSPK bands ➡️ SNAP fails with a predictor issue ➡️ downgrade sertit to 1.19.1:
✅  `pip install sertit[full]==1.19.1`

**Comment by TK12331** [comment]
> Ouuuh yeah, so relieved! 🚀 Sorry you arrived at a bad time 😅
> 
> This will all be resolved in 0.18.0...

I'm looking forward to it hhhh.
